gammapy/estimators/points/sed.py

```diff
@@ -136,13 +136,31 @@ def run(self, datasets):
         if not datasets.energy_axes_are_aligned:
             raise ValueError("All datasets must have aligned energy axes.")
 
-        if "TELESCOP" in datasets.meta_table.colnames:
-            telescopes = datasets.meta_table["TELESCOP"]
-            if not len(np.unique(telescopes)) == 1:
-                raise ValueError(
-                    "All datasets must use the same value of the"
-                    " 'TELESCOP' meta keyword."
-                )
+        if (
+            len(datasets) > 1
+            and datasets[0].meta_table is not None
+            and any(
+                dataset.meta_table["TELESCOP"].shape
+                != datasets[0].meta_table["TELESCOP"].shape
+                for dataset in datasets
+            )
+        ):
+            for dataset in datasets:
+                if "TELESCOP" in dataset.meta_table.colnames:
+                    telescopes = dataset.meta_table["TELESCOP"]
+                    if not len(np.unique(telescopes)) == 1:
```

Comment:
I will extract this part before the first if. In this way you don't have duplicate code

Reply:
The code is not duplicated, note that we extract from `dataset.meta_table` here but below it is from `datasets.meta_table`

Reply:
yes but this part is duplicate:

```python
telescopes = datasets.meta_table["TELESCOP"]
                if not len(np.unique(telescopes)) == 1:
                    raise ValueError(
                        "All datasets must use the same value of the"
                        " 'TELESCOP' meta keyword."
                    )
```